converter.go

```diff
@@ -570,6 +806,18 @@ func arrowToValue(rowIdx int, srcColumnMeta execResponseRowType, srcValue arrow.
 	return nil, fmt.Errorf("unsupported data type")
 }
 
+func mapStructuredArrayNativeArrowRows[T any](offsets []int32, rowIdx int, values arrow.Array, f func(j int) (T, error)) (snowflakeValue, error) {
+	arr := make([]T, offsets[rowIdx+1]-offsets[rowIdx])
+	for j := offsets[rowIdx]; j < offsets[rowIdx+1]; j++ {
+		v, err := f(int(j))
```

Comment:
I think this is better. I would just give this passed function more meaningful name instead of `f`.